### ISSUE TITLE
Text in form fields overlaps label when scrolled

### ISSUE THREAD

**Issue by seanh** [hypothesis/h#3764]
![screenshot from 2016-08-23 18-17-06](https://cloud.githubusercontent.com/assets/22498/17902178/2aff1bf6-695e-11e6-832c-43e5a0ea70eb.png)

Of course that description field is too small (the proposed max length for a description is 250 chars, and that description field is too small to contain 250 chars without scrolling), but that's a separate issue. This issue is to report that when any field does get a scrollbar within, the text overlaps the label.

**Comment by sean-roberts** [comment]
That's pretty tricky. Nothing immediately obvious that can be done to fix the overlapping as the css is. I think we just need to rework the css so that the border and padding is on the container element, not the input field itself.

**Comment by sean-roberts** [comment]
Unless someone has other ideas. I moved the concern of padding and borders onto the container element (.form-input). The caveat with this is focus pseudo state is not available. We can use JS to add `is-editing` class to the container and we will style that with the think-border styling. Obviously that means styling for that particular state won't reflect in non js environments. I think this is an ok compromise. 

I already styled the is-editing class be thick-container. I saw other references to is-editing in the css already. That said, I don't think that is currently working from the js side. @robertknight thoughts on all of this?

**Comment by robertknight** [comment]
> Unless someone has other ideas. I moved the concern of padding and borders onto the
> container element (.form-input). The caveat with this is focus pseudo state is not available.

Short answer: Makes sense. This is what @seanh and I discussed doing the last time we looked at this.

Longer answer: As you've seen, the way input fields are currently implemented is to give the main `<input>` element sufficient padding to provide space for its labels, error messages etc. and then add the focus ring to it using `:focus`. The advantage is that you get all of the standard focus/interaction functionality for free. The downside is issues like this. The alternative, putting the padding/borders on the container and implementing focus handling in JS adds some complexity but would avoid problems like this.

As long as the user can easily interact with the field (navigate with keyboard/mouse, identify which is focused) in a non-JS env, it is OK if the styling looks a little different.

> I already styled the is-editing class be thick-container. I saw other references to is-editing
> in the css already. That said, I don't think that is currently working from the js side.
> @robertknight thoughts on all of this?

I'd suggest we come back to this once https://github.com/hypothesis/h/pull/3911 is merged, because that changes the states for input fields.
